### PR TITLE
Add feature flag for cosmetic filtering in child frames

### PR DESCRIPTION
**Pull request by antonok-edm**
Resolves https://github.com/brave/brave-browser/issues/23891

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by pes10k — APPROVED** [review]
LGTM, but should be merged with the feature disabled in griffin

File: components/brave_shields/common/features.cc [hunk 1]
```diff
@@ -32,6 +32,9 @@ const base::Feature kBraveAdblockCookieListDefault{
 const base::Feature kBraveAdblockCosmeticFiltering{
     "BraveAdblockCosmeticFiltering",
     base::FEATURE_ENABLED_BY_DEFAULT};
+const base::Feature kBraveAdblockCosmeticFilteringChildFrames{
+    "BraveAdblockCosmeticFilteringChildFrames",
+    base::FEATURE_ENABLED_BY_DEFAULT};
```
Comment (pes10k) [inline components/brave_shields/common/features.cc:37]: until the perf issue is sorted out, would be better to ship disabled by default

cc @bbondy

**Comment by antonok-edm** [comment]
Updated to disable the feature by default

**Review by atuchin-m — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.43.31 Chromium: 103.0.5060.114 (Official Build) nightly (64-bit)
-- | --
Revision | a1c2360c5b02a6d4d6ab33796ad8a268a6128226-refs/branch-heads/5060@{#1124}
OS | Windows 11 Version 21H2 (Build 22000.795)
```

Ensured that `#brave-adblock-cosmetic-filtering-child-frames` was present via `brave://flags` and was disabled by running through the `Local frame containing a low-text ad` check via https://dev-pages.bravesoftware.com/filtering/text-ads.html as per the following:

**`Example`** | **`Example`**
----------------|----------------
flagPresent | disabledDefault

Enabled `#brave-adblock-cosmetic-filtering-child-frames` via `brave://flags` and ensured there wasn't any text appearing via the `Local frame containing a low-text ad` check using https://dev-pages.bravesoftware.com/filtering/text-ads.html as per the following:

**`Example`** | **`Example`**
----------------|----------------
enabledChildFrame | enabledExample

**Pull request merged**